src/main/java/org/ladocuploader/app/cli/SubmissionTransfer.java

```diff
@@ -0,0 +1,121 @@
+    this.fileRepositoryService = fileRepositoryService;
+    this.fileRepository = fileRepository;
+    this.pdfService = pdfService;
+    this.pgpEncryptor = pgpEncryptor;
+    this.encryptor = encryptor;
+    this.ftpsClient = ftpsClient;
+  }
+
+  public void transferSubmissions() {
+    // TODO get submissions to transfer
+
+    String batchIndex = "5000"; // TODO Prob needs a counter
+    String zipFileName = batchIndex + ".zip";
+
+    log.info(String.format("Beginning transfer of batch %s", batchIndex));
+
+    // for each, add to zipfile
+    List<Submission> submissionsBatch = new ArrayList<>(); // TODO Get from transmission table
+    try (FileOutputStream docMeta = new FileOutputStream(batchIndex + ".txt");
+         FileOutputStream baos = new FileOutputStream(zipFileName);
+         ZipOutputStream zos = new ZipOutputStream(baos)) {
+      for (Submission submission : submissionsBatch) {
+        String subfolder = "1"; // TODO Something unique
+        try {
+          if ("laDigitalAssister".equals(submission.getFlow())) {
+            log.info("Generate applicant summary");
+            byte[] file = pdfService.getFilledOutPDF(submission); // TODO Handle generate-pdf crashes
+            String fileName = String.format("%s_application.pdf", subfolder);
+            zos.putNextEntry(new ZipEntry(subfolder));
+            ZipEntry entry = new ZipEntry(subfolder + fileName);
+            entry.setSize(file.length);
+            zos.putNextEntry(entry);
+            zos.write(file);
+            zos.closeEntry();
+            byte[] metaEntry = generateMetaDataEntry(batchIndex, subfolder, fileName, "APP-OFS 4APP", submission);
+            docMeta.write(metaEntry);
+
+            log.info("Adding uploaded docs");
+            List<UserFile> userFiles = fileRepositoryService.findAllBySubmission(submission);
+            for (UserFile userFile : userFiles) {
+              String docUploadFilename = "somethingunique"; // TODO this can be anything
+              ZipEntry docEntry = new ZipEntry(subfolder + docUploadFilename);
+              docEntry.setSize(userFile.getFilesize().longValue());
+              zos.putNextEntry(docEntry);
+
+              CloudFile docFile = fileRepository.get(userFile.getRepositoryPath());
+              zos.write(docFile.getFileBytes());
+              zos.closeEntry();
+
+              // write doc metadata - TODO add docType
+              metaEntry = generateMetaDataEntry(batchIndex, userFile.getOriginalName(), "", "", submission);
+              docMeta.write(metaEntry);
+            }
+          }
+        } catch (Exception e) {
+          log.error("Error generating file collection for submission ID {}", submission.getId(), e);
+        }
+        zos.close();
+        baos.close();
+        docMeta.close();
+
+        // Encrypt and transfer
+        byte[] data = pgpEncryptor.signAndEncryptPayload(zipFileName);
+        ftpsClient.uploadFile(zipFileName, data);
```

Comment:
Should we delete the zipfile here, once it's in the byte array?   Mostly concerned about disk space filling up in the container.

Reply:
Fixed